<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>947</width>
    <height>812</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Signal Analysator (SET)</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icon.png</normaloff>icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="0" column="0">
     <widget class="QGroupBox" name="main_layout">
      <property name="title">
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout_9">
       <item row="0" column="0" colspan="2">
        <widget class="QFrame" name="plot_frame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QTabWidget" name="filter">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="control_tab">
       <attribute name="title">
        <string>Control</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0" colspan="2">
         <widget class="QFrame" name="devices_frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_17"/>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QPushButton" name="open_close_stream">
          <property name="text">
           <string>Attach to task</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>28</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QFrame" name="control_frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_16"/>
         </widget>
        </item>
        <item row="4" column="0">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>542</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="advanced_tab">
       <attribute name="title">
        <string>Advanced</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QFrame" name="modules_frame">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_13">
           <item row="0" column="0">
            <widget class="QGroupBox" name="afc_frame">
             <property name="title">
              <string>AFC window</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" name="gridLayout_7">
              <item row="0" column="0">
               <widget class="QRadioButton" name="btn_short_range">
                <property name="text">
                 <string>Short range</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QRadioButton" name="btn_wide_range">
                <property name="text">
                 <string>Wide range</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="lbl_window_function">
                <property name="text">
                 <string>Window function</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QComboBox" name="cmb_window_function">
                <item>
                 <property name="text">
                  <string>None</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Bartlett</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Blackman</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>BlackmanStd</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>BlackmanOpt</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Hamming</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Hann</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="lbl_window_alpha">
                <property name="text">
                 <string>Window alpha</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QDoubleSpinBox" name="spnbox_window_alpha">
                <property name="maximum">
                 <double>2.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.160000000000000</double>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="lbl_num_peaks">
                <property name="text">
                 <string>Number of peaks</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QSpinBox" name="box_num_peaks"/>
              </item>
             </layout>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QGroupBox" name="main_window_properties">
             <property name="title">
              <string>Processing data method</string>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" name="gridLayout_5">
              <item row="3" column="0">
               <widget class="QCheckBox" name="phase_offset">
                <property name="text">
                 <string>Substitution of the signal phase</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QRadioButton" name="chk_resampling_method">
                <property name="text">
                 <string>Signal resampling</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QRadioButton" name="chk_sig_transform">
                <property name="text">
                 <string>Signal transformation</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QCheckBox" name="chk_reject_data">
                <property name="text">
                 <string>Reject data</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QRadioButton" name="chk_delta_method">
                <property name="text">
                 <string>Default method</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QFrame" name="offset_frame">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_15">
           <item row="0" column="0">
            <widget class="QwtTextLabel" name="offset_label">
             <property name="plainText">
              <string>Phase offset</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QwtSlider" name="offset_slider">
             <property name="lowerBound">
              <double>-500.000000000000000</double>
             </property>
             <property name="upperBound">
              <double>500.000000000000000</double>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="trough">
              <bool>true</bool>
             </property>
             <property name="groove">
              <bool>true</bool>
             </property>
             <property name="borderWidth">
              <number>2</number>
             </property>
             <property name="spacing">
              <number>1</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QFrame" name="afc_progress_frame">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="1" column="0">
            <widget class="QProgressBar" name="progressBar">
             <property name="maximum">
              <number>0</number>
             </property>
             <property name="value">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QwtTextLabel" name="afc_progress_bar">
             <property name="plainText">
              <string>AFC extract progress</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0">
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>13</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="denoised_tab">
       <attribute name="title">
        <string>Additional</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="0" column="0">
         <widget class="QFrame" name="denoised_frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_19"/>
         </widget>
        </item>
        <item row="1" column="0">
         <spacer name="verticalSpacer_6">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>28</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0">
         <widget class="QFrame" name="record_frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_18"/>
         </widget>
        </item>
        <item row="3" column="0">
         <spacer name="verticalSpacer_5">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>28</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="4" column="0">
         <widget class="QGroupBox" name="chk_frame">
          <property name="title">
           <string>Enable check stream</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="lbl_num_chk_buff">
             <property name="text">
              <string>Buffers checked</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLCDNumber" name="lcd_num_chk_buff">
             <property name="styleSheet">
              <string notr="true">color: rgb(0, 255, 0);
background-color: rgb(0, 0, 0);</string>
             </property>
             <property name="digitCount">
              <number>9</number>
             </property>
             <property name="segmentStyle">
              <enum>QLCDNumber::Flat</enum>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lbl_num_err_buff">
             <property name="text">
              <string>Invalid buffers</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLCDNumber" name="lcd_num_err_buff">
             <property name="styleSheet">
              <string notr="true">color: rgb(0, 255, 0);
background-color: rgb(0, 0, 0);</string>
             </property>
             <property name="digitCount">
              <number>9</number>
             </property>
             <property name="segmentStyle">
              <enum>QLCDNumber::Flat</enum>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lbl_duration_tm">
             <property name="text">
              <string>Duration of check</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLCDNumber" name="lcd_duration_tm">
             <property name="styleSheet">
              <string notr="true">color: rgb(0, 255, 0);
background-color: rgb(0, 0, 0);</string>
             </property>
             <property name="digitCount">
              <number>9</number>
             </property>
             <property name="segmentStyle">
              <enum>QLCDNumber::Flat</enum>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lbl_lost_session">
             <property name="text">
              <string>Lost per session</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLCDNumber" name="lcd_lost_session">
             <property name="styleSheet">
              <string notr="true">color: rgb(0, 255, 0);
background-color: rgb(0, 0, 0);</string>
             </property>
             <property name="digitCount">
              <number>9</number>
             </property>
             <property name="segmentStyle">
              <enum>QLCDNumber::Flat</enum>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="lbl_lost_period">
             <property name="text">
              <string>Lost per period</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLCDNumber" name="lcd_lost_period">
             <property name="styleSheet">
              <string notr="true">color: rgb(0, 255, 0);
background-color: rgb(0, 0, 0);</string>
             </property>
             <property name="digitCount">
              <number>9</number>
             </property>
             <property name="segmentStyle">
              <enum>QLCDNumber::Flat</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="5" column="0">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>667</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>947</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLanguage"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionLanguage">
   <property name="text">
    <string>Language</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QwtSlider</class>
   <extends>QWidget</extends>
   <header>qwt_slider.h</header>
  </customwidget>
  <customwidget>
   <class>QwtTextLabel</class>
   <extends>QFrame</extends>
   <header>qwt_text_label.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>446</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_short_range</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setAFCWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>784</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_wide_range</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setAFCWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>784</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>afc_frame</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setAFCWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>784</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>open_close_stream</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>streamOpenClose()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>780</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setSignalParams()</slot>
  <slot>streamOpenClose()</slot>
  <slot>setDivider(int)</slot>
  <slot>setScanFrequency(int)</slot>
  <slot>setRefreshRate(double)</slot>
  <slot>setFilterWindow()</slot>
  <slot>setMedianK(int)</slot>
  <slot>setPhaseOffset(int)</slot>
  <slot>changeExchangeBehavior()</slot>
  <slot>setStreamType(QString)</slot>
  <slot>showRecord()</slot>
  <slot>setStreamWindow()</slot>
  <slot>setAFCWindow()</slot>
 </slots>
</ui>
